### PR TITLE
feat: refactor create collection TDE-528

### PR DESCRIPTION
**Pull request by MDavidson17**
partial completed of TDE-528 (still need to update workflow)

`collection_from_items.py` and most of `create_stac.py` have been deleted as they are not used/duplicate code.

the script has been altered to:
- create multiple collections during standardise_validate
- flatten_collection.py has been created to take a list of collections (from aws-list) and merge them if they have the same collection_id
    - it merges links and recalculates the bbox and interval

### CODE REVIEW

File: scripts/flatten_collections.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
+import argparse
```
Comment (paulfouquet) [inline scripts/flatten_collections.py:1]: Would `merge_collections.py` make more sense for this script?

File: scripts/flatten_collections.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+            continue
+
+        if not collection:
+            collection = ImageryCollection(
+                title=partial_stac["title"], description=partial_stac["description"], collection_id=arguments.collection_id
+            )
+            collection.stac = partial_stac
+            continue
+
+        # merge links
```
Comment (paulfouquet) [inline scripts/flatten_collections.py:53]: Would be nice put these 3 following merge into separates function and have some unit tests for them, simulating a list of `partial_stac` to be merged together.

File: scripts/flatten_collections.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+def main() -> None:
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--source", dest="source", nargs="+", required=True)
+    parser.add_argument("--collection_id", dest="collection_id", required=True)
+    parser.add_argument("--destination", dest="destination directory", required=True)
+
+    arguments = parser.parse_args()
+
+    source = format_source(arguments.source)
+
```
Comment (paulfouquet) [inline scripts/flatten_collections.py:22]: code style: few blank lines that I don't think are necessary. Blank lines should be used to indicate logical sections.

**Review by paulfouquet — CHANGES_REQUESTED** [review]
Looks good to me. Only a couple of minor comments and would be nice to have few unit tests for the merging functions.

**Review by amfage — APPROVED** [review]
:+1:

**Review by paulfouquet — CHANGES_REQUESTED** [review]

File: scripts/conftest.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
+from typing import Generator
+
+import pytest
+
+from scripts.stac.imagery.collection import ImageryCollection
+
+
+@pytest.fixture(autouse=True)
```
[the remark this answers is not in the document]
Comment (MDavidson17) [reply scripts/conftest.py:8]: Yea, I tried this first but got mypy issues but I'll take a closer look

**Comment by MDavidson17** [comment]
superceeded by: `https://github.com/linz/topo-imagery/pull/219`

**Pull request closed without merging**